backend/core/api_serializers/author_api_serializer.py

```diff
@@ -30,55 +33,46 @@ def get_single_author(self, author_id):
         author_data = {}
         try:
             author = author_serializer.get_author_by_id(author_id)
-            author_data = self.get_author_data(author)
-        except ValueError:
+        except ValidationError:
             return None
-        return author_data
+        
+        serializer = AuthorSerializer(author)
+
+        return serializer.data
     
     def get_all_authors(self, page=None, size=None):
-
         authors = Author.objects.all()
         
-        result_dict = {}
-        result_dict["type"] = "authors"
-
-        authors_list = []
+        if page and size:
+            paginator = Paginator(authors, size)
+            authors = paginator.get_page(page)
 
-        for author in authors:
-            curr_author_data = self.get_author_data(author)
-            authors_list.append(curr_author_data)
+        authors_serializer = AuthorSerializer(authors, many=True)
         
-        if page and size:
-            paginated_authors = utils.paginate_list(authors_list, page, size)
-            
-            result_dict["page"] = page
-            result_dict["size"] = size
-            result_dict["items"] = paginated_authors
-        else:
-            result_dict["items"] = authors_list
+        serializer = AllAuthorSerializer(data={
+                        'type': 'authors',
+                        'page': page,
+                        'size': size,
+                        'items': authors_serializer.data
```

Comment:
Are we setting the default values for page and size? Because generally these parameters are optional so we should be careful about this.

Reply:
Yeah, if its null, we handle that too